Unicode/is_Ligature.c

```diff
@@ -0,0 +1,212 @@
+  0x2155, 0x2156, 0x2157, 0x2158, 0x2159, 0x215a, 0x215b, 0x215c,
+  0x215d, 0x215e, 0x2189};
+
+const uint16 ____fraction16[] = {
+  0x0b72, 0x0b73, 0x0b74, 0x0b75, 0x0b76, 0x0b77, 0x0c78, 0x0c79,
+  0x0c7a, 0x0c7b, 0x0c7c, 0x0c7d, 0x0c7e, 0x0d73, 0x0d74, 0x0d75,
+  0x2044, 0x215f, 0x2cfd, 0xa830, 0xa831, 0xa832, 0xa833, 0xa834,
+  0xa835};
+
+const uint32 ____fraction32[] = {
+  0x00010e7b, 0x00010e7c, 0x00010e7d, 0x00010e7e};
+
+static int compare_codepoints16(const void *uCode1, const void *uCode2) {
+    const uint16 *cp1 = (const uint16 *)(uCode1);
+    const uint16 *cp2 = (const uint16 *)(uCode2);
+    return( (*cp1 < *cp2) ? -1 : ((*cp1 == *cp2) ? 0 : 1) );
+}
+
+static int compare_codepoints32(const void *uCode1, const void *uCode2) {
+    const uint32 *cp1 = (const uint32 *)(uCode1);
+    const uint32 *cp2 = (const uint32 *)(uCode2);
+    return( (*cp1 < *cp2) ? -1 : ((*cp1 == *cp2) ? 0 : 1) );
+}
+
+int LigatureCount(void) {
+    return( 509 );
+}
+
+int VulgarFractionCount(void) {
+    return( 19 );
+}
+
+int OtherFractionCount(void) {
+    return( 29 );
+}
+
+int FractionCount(void) {
+    return( 48 );
+}
+
+int32 Ligature_get_U(int n) {
+    if ( n<0 || n>=509 )
+	return( -1 );
+    return( (int32)(____ligature16[n]) );
+}
+
+int32 VulgFrac_get_U(int n) {
+    if ( n<0 || n>=19 )
+	return( -1 );
+    return( (int32)(____vulgfrac16[n]) );
+}
+
+int32 Fraction_get_U(int n) {
+    if ( n<0 || n>=29 )
+	return( -1 );
+    if ( n<25 )
+	return( (int32)(____fraction16[n]) );
+    else
+	return( (int32)(____fraction32[n-25]) );
+}
+
+int Ligature_find_N(uint32 uCode) {
+    uint16 uCode16, *p16;
+    int n=-1;
+
+    if ( uCode<0x132 || uCode>0xfefc || isligorfrac(uCode)==0 )
+	return( -1 );
+    uCode16 = uCode;
+    p16 = (uint16 *)(bsearch(&uCode16, ____ligature16, 509, \
+				sizeof(uint16), compare_codepoints16));
+    if ( p16 ) n = p16 - ____ligature16;
+    return( n );
+}
+
+int VulgFrac_find_N(uint32 uCode) {
+    uint16 uCode16, *p16;
+    int n=-1;
+
+    if ( uCode<0xbc || uCode>0x2189 || isligorfrac(uCode)==0 )
+	return( -1 );
+    uCode16 = uCode;
+    p16 = (uint16 *)(bsearch(&uCode16, ____vulgfrac16, 19, \
+				sizeof(uint16), compare_codepoints16));
+    if ( p16 ) n = p16 - ____vulgfrac16;
+    return( n );
+}
+
+int Fraction_find_N(uint32 uCode) {
+    uint16 uCode16, *p16;
+    uint32 *p32;
+    int n=-1;
+
+    if ( uCode<0xb72 || uCode>0x10e7e || (uCode<65536 && isligorfrac(uCode)==0) )
+	return( -1 );
+    uCode16 = uCode;
+    if ( uCode<25 ) {
+	uCode16 = uCode;
+	p16 = (uint16 *)(bsearch(&uCode16, ____fraction16, 25, \
+				sizeof(uint16), compare_codepoints16));
+	if ( p16 ) n = p16 - ____fraction16;
+    } else {
+	p32 = (uint32 *)(bsearch(&uCode, ____fraction32, 4, \
+				sizeof(uint32), compare_codepoints32));
+	if ( p32 ) n = p32 - ____fraction32 + 25;
+    }
+    return( n );
+}
+
+/* Codepoints whose names match the regular expression (yes==0) */
```

Comment:
So instead of saying that we are looking for things that match "the" regular expression yes == 0, which I did not understand even after looking at the functions, we would say either that we are looking for non-vulgar fractions or that we are looking for things whose names include FRACTION but not VULGAR.